opencv-kinfu-samples/CMakeLists.txt

```diff
@@ -0,0 +1,10 @@
+cmake_minimum_required(VERSION 3.10)
+add_executable(kinfu-example main.cpp)
+
+find_package(k4a 1.2.0 QUIET)
+include_directories(${K4A_INCLUDE_DIRS})
+
+find_package(OpenCV)
+include_directories(${OpenCV_INCLUDE_DIRS})
+
+target_link_libraries(kinfu-example PRIVATE k4a::k4a opencv_world)
```

Comment:
>opencv_world [](start = 53, length = 12)

I haven't had a chance to install opencv on my linux box, especially also need the module implemented kinfu from opencv_contrib repo, just curious, is opencv_world enough? or is there some opencv_contrib_world as well? It might be very helpful that to describe the Linux opencv dependency installation steps in the README.md file, similar as what we did for windows, would you mind add that? I probably can try build and install opencv contrib on my Ubuntu box to test it out when I get some time.

Reply:
So the implementation of `kinfu` is in `rgbd` module (in `opencv_contrib`). If you notice [RGBD module's `CMakeLists.txt` file](https://github.com/opencv/opencv_contrib/blob/master/modules/rgbd/CMakeLists.txt), you will see that its necessary dependencies are loaded by a macro. I added `opencv_world` just to make sure that other modules could be included for safety and possible addition of some trivial opencv features in future doesn't require change in `CMakeLists.txt`.

Regarding the `opencv_contrib_world` it is outdated now and is fixed by https://github.com/opencv/opencv/pull/8044. So, basically, `opencv_world` takes care of the contrib modules too. 

Adding only `opencv_rgbd` (compiled with `opencv_viz`)is technically the best thing at this point and works well on my system. Also because, `opencv_world` is often too tedious to build successfully. 

Regarding the Linux dependency installation steps in `README.md`, I think I'll do documentation improvements in different PR in few days.

Reply:
@rabbitdaxi @yijiew, I have also added some guide for Linux dependency like you said fcc4934622baf489a71c9274f121ddd11f679501. Didn't create a separate PR for that. Should be okay.

Reply:
Thank you for doing so~!

---
In reply to: [372307869](https://github.com/microsoft/Azure-Kinect-Samples/pull/40#discussion_r372307869) [](ancestors = 372307869)

Reply:
I still did not get a chance to try verify it on Linux. Given you verified on your Linux box and by looking at the code all good to me, we can get this in.

---
In reply to: [367586174](https://github.com/microsoft/Azure-Kinect-Samples/pull/40#discussion_r367586174) [](ancestors = 367586174)